metaflow/plugins/datatools/s3/s3.py

```diff
@@ -1601,10 +1607,13 @@ def try_s3_op(last_ok_count, pending_retries, out_lines, inject_failures):
                     try:
                         debug.s3client_exec(cmdline + addl_cmdline)
                         # Run the operation.
+                        env = os.environ.copy()
+                        tracing.inject_tracing_vars(env)
```

Comment:
ok to inject the whole env here? no concern of exposing secrets?

Reply:
Do we foreseen any risk here?
cc @oavdeev as this is copied from the initial commit

Reply:
inject_tracing_vars adds tracing vars into dictionary passed to it